doc/rst/language/spec/procedures.rst

```diff
@@ -576,70 +576,93 @@ Abstract Intents
-``unmanaged class`` ``const in``     ``const in``           
-``atomic``          ``const ref``    ``ref``                
-``single``          ``const ref``    ``ref``                
-``sync``            ``const ref``    ``ref``                
-``string``          ``const ref``    ``const ref``          
-``bytes``           ``const ref``    ``const ref``          
-``record``          ``const ref``    ``const ref``           see :ref:`Default_Intent_for_Arrays_and_Record_this`
-``union``           ``const ref``    ``const ref``          
-``dmap``            ``const ref``    ``const ref``          
-``domain``          ``const ref``    ``const ref``          
-array               ``const ref``    ``ref`` / ``const ref`` see :ref:`Default_Intent_for_Arrays_and_Record_this`
-tuple               per element      per element             see :ref:`Tuple_Argument_Intents`
-=================== ================ ======================= ====================================================
-
 .. _The_Const_Intent:
 
 The Const Intent
 ^^^^^^^^^^^^^^^^
 
-The ``const`` intent specifies the intention that the function will not
-and cannot modify the formal argument within its dynamic scope. Whether
-the actual argument will be passed by ``const in`` or ``const ref``
-intent depends on its type. In general, small values, such as scalar
-types, will be passed by ``const in``; while larger values, such as
-domains and arrays, will be passed by ``const ref`` intent. The
-:ref:`Abstract_Intents_Table` earlier in this sub-section lists the
-meaning of the const intent for each type.
+The ``const`` intent specifies that the function will not and cannot
+modify the formal argument within its dynamic scope. Whether ``const``
+is interpreted as ``const in`` or ``const ref`` intent depends on the
+argument type.  Generally, small values, such as scalar types, will be
+passed by ``const in``; while larger values, such as domains and
+arrays, will be passed by ``const ref`` intent. The
+:ref:`Abstract_Intents_Table` below lists the meaning of the ``const``
+intent for each type.
 
 .. _The_Default_Intent:
 
 The Default Intent
 ^^^^^^^^^^^^^^^^^^
 
-When no intent is specified for a formal argument, the *default intent*
-is applied. It is designed to take the most natural/least surprising
-action for the argument, based on its type.
-The :ref:`Abstract_Intents_Table` earlier in this sub-section lists the
-meaning of the default intent for each type.
+When no intent is specified for a formal argument, the *default
+intent* is applied.  It is designed to take the most natural/least
+surprising action for the argument, based on its type.  In practice,
+this is ``const`` for most types (as defined by
+:ref:`The_Const_Intent`) to avoid surprises for programmers coming
+from languages where everything is passed by ``in`` or ``ref`` intent
+by default.  Exceptions are made for types where modification is
+considered part of its nature, such as types used for synchronization
```

Comment:
a singular/plural mismatch; "types ... part of its" maybe should be "types ... part of their"

Reply:
Is it "their nature" or "their natures"?

Reply:
I think it's "for types where modification is considered part of their nature"